src/main/java/com/salesforce/storm/spout/dynamic/DynamicSpout.java

```diff
@@ -395,6 +395,14 @@ public boolean hasVirtualSpout(final VirtualSpoutIdentifier spoutIdentifier) {
     }
 
     /**
+     * Get the total number of virtual spouts in the coordinator.
+     * @return total number of virtual spouts in the coordinator.
+     */
+    public int getTotalVirtualSpouts() {
```

Comment:
This is crossing the thread barrier, may want to validate that its ok/safe to do so.  We may also want to start denoting when we do this